gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go

```diff
@@ -165,24 +165,16 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 		return nil, fmt.Errorf("glusterblock: failed to create volume: %v", err)
 	}
 
-	if len(vol.Portals) == 1 && vol.Portals[0] != "" {
-		vol.TargetPortal = vol.Portals[0]
-		vol.Portals = nil
-	} else {
+	//Sort Target Portal from portal.
+	sortErr := p.sortTargetPortal(vol)
+	if sortErr != nil {
+		return nil, fmt.Errorf("glusterblock: failed to fetch Target Portal: %v", err)
+	}
 
-		portals := vol.Portals
-		vol.Portals = nil
-		for _, v := range portals {
-			if v != "" && vol.TargetPortal == "" {
-				vol.TargetPortal = v
-				continue
-			} else {
-				vol.Portals = append(vol.Portals, v)
-			}
-		}
+	if vol.TargetPortal == "" || vol.Iqn == "" {
+		return nil, fmt.Errorf("glusterblock: Target portal/IQN is nil")
```

Comment:
Yeah, those are just place holders.  Also above check is added here to make sure we are continuing when these fields are available. Less chance to mess up in between though.